src/smartpeak/source/core/Utilities.cpp

```diff
@@ -605,4 +606,45 @@ namespace SmartPeak
 
     return n;
   }
+
+  std::string Utilities::getLogFilepath(const std::string& filename)
+  {
+      std::string path = "";
+      bool flag = false;
+      char const* logs = getenv("SMARTPEAK_LOGS");
+      char const* appdata = getenv("LOCALAPPDATA");
+      char const* home = getenv("HOME");
+      if (logs || appdata || home)
+      {
+          auto logdir = fs::path{};
+          if (logs)
+              logdir = fs::path{ logs };
+          else if (appdata)
+              logdir = fs::path{ appdata } / "SmartPeak";
+          else if (home)
+              logdir = fs::path{ home } / ".SmartPeak";
+
+          try
+          {
+              // Creates directory tree if doesn't exist:
+              flag = fs::create_directories(logdir);
```

Comment:
nice work, adding a mechanism to check whether the folder is still writable by SmartPeak in case it exists would be very beneficial :)

Reply:
I addressed this in new commits thx.